src/EPR.Payment.Service.Data.UnitTests/Repositories/PaymentsRepositoryTests.cs

```diff
@@ -0,0 +1,199 @@
+﻿using AutoFixture.MSTest;
+using EPR.Payment.Service.Common.Data;
+using EPR.Payment.Service.Common.Data.Repositories;
+using EPR.Payment.Service.Common.UnitTests.Mocks;
+using EPR.Payment.Service.Common.UnitTests.TestHelpers;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Moq.EntityFrameworkCore;
+using System.Data.Entity;
+
+namespace EPR.Payment.Service.Data.UnitTests.Repositories
+{
+    [TestClass]
+    public class PaymentsRepositoryTests
+    {
+        private Mock<DbSet<Common.Data.DataModels.Payment>> _paymentMock;
+        private Mock<DbSet<Common.Data.DataModels.Lookups.PaymentStatus>> _paymentStatusMock;
+        public PaymentsRepositoryTests()
+        {
+            _paymentMock = MockIPaymentRepository.GetPaymentMock();
+            _paymentStatusMock = MockIPaymentRepository.GetPaymentStatusMock(true);
+        }
+
+        [TestMethod]
+        [AutoMoqData]
+        public async Task InsertPaymentStatusAsync_ValidInput_ShouldReturnGuid(
+            [Frozen] Mock<AppDbContext> _dataContextMock, 
+            [Greedy] PaymentsRepository _mockPaymentsRepository)
+        {
+            //Arrange
+            _dataContextMock.Setup(i => i.Payment).ReturnsDbSet(_paymentMock.Object);
+            _mockPaymentsRepository = new PaymentsRepository(_dataContextMock.Object);
+
+            var newId = Guid.NewGuid();
+            var userId = Guid.NewGuid();
+            var organisationId = Guid.NewGuid();
+            var request = new Common.Data.DataModels.Payment
+            {
+                Id = newId,
+                UserId = userId,
+                OrganisationId = organisationId,
+            };
+
+            //Act
+            Guid result = await _mockPaymentsRepository.InsertPaymentStatusAsync(request);
+
+
+            //Assert
```

Comment:
Use assertion scope if you are doing more than one assertions in a single test;
            //Assert
            using (new AssertionScope())
            {
                result.Should().NotBe(Guid.Empty);
                _dataContextMock.Verify(c => c.Payment.Add(It.Is<Common.Data.DataModels.Payment>(s => s.UserId == userId && s.OrganisationId == organisationId)), Times.Once());
                _dataContextMock.Verify(c => c.SaveChangesAsync(default), Times.Once);
            }

Reply:
resolved